File: javascript/rna-transcription/rna-transcription.js
var DnaTranscriber = function () {
  this.dnaToRna = {
    'C' : 'G',
    'G' : 'C',
    'A' : 'U',
    'T' : 'A'
  };
}

DnaTranscriber.prototype.toRna = function (dna) {
  var dnaToRna = this.dnaToRna;
  var rna = dna.split('').map(function(char){
      return dnaToRna[char];
  })
  if (rna.includes(undefined)) { throw new Error('Invalid input') }
  return rna.join('');
};

module.exports = DnaTranscriber;

File: ruby/rna-transcription/rna_transcription.rb
# RNA Transcription
class Complement
  def self.of_dna(strand)
    rna = strand.split('').map { |dna| convert(dna) }
    rna.include?(nil) ? '' : rna.join('')
  end

  def self.convert(dna)
    case dna
    when 'G'
      'C'
    when 'C'
      'G'
    when 'T'
      'A'
    when 'A'
      'U'
    end
  end
end

module BookKeeping
  VERSION = 4
end
